### PR TITLE
Smart fees for wallet CreateTransaction

### PR DESCRIPTION
**Pull request by rnicoll**
Dogecoin equivalent of bitcoin/bitcoin#4250

### CODE REVIEW

**Comment by rnicoll** [comment]
For discussion and review, do not merge yet. Will test further this afternoon.

File: src/wallet.h [hunk 1]
```diff
@@ -25,12 +25,14 @@
...
 // Settings
 extern CFeeRate payTxFee;
+extern unsigned int nTxConfirmTarget;
 extern bool bSpendZeroConfChange;
 
 // -paytxfee default
 static const int64_t DEFAULT_TRANSACTION_FEE = 0;
 // -paytxfee will warn if called with a higher fee than this amount (in satoshis) per KB
 static const int nHighTransactionFeeWarning = 25 * COIN;
+static const unsigned int MAX_FREE_TRANSACTION_CREATE_SIZE = 1000;
```
Comment (patricklodder) [inline src/wallet.h]: propose: MAX_FREE_TRANSACTION_CREATE_SIZE = 0;

**Comment by rnicoll** [comment]
That should be it ready to merge now unless there's any other feedback.

File: doc/release-notes.md [hunk 2]
```diff
@@ -1,6 +1,26 @@
...
+high a priority) transactions require to be confirmed quickly. The default
+settings will create transactions that confirm quickly; see the new
+'txconfirmtarget' setting to control the tradeoff between fees and
+confirmation times.
+
+Prior releases used hard-coded fees (and priorities), and would
+sometimes create transactions that took a very long time to confirm.
+
+
+New Command Line Options
```
Comment (langerhans) [inline doc/release-notes.md]: Reminder: Rework the release notes at release time, as this is not actually in with this PR

**Comment by langerhans** [comment]
I'll give this a try soon, but looks good so far.

**Comment by langerhans** [comment]
Seems like it doesn't calculate the dust fee correctly. Sent this tc on mainnet just now: 3d6ec3ae2aca3ae0a6c65074fd8ee888cd7ed262f2cbaa25d33861989324a14e  

It should have been 2 DOGE fee because of the dust output, no?

**Comment by rnicoll** [comment]
Think so, but can't see how it would miss it, and more I can't see how it was relayed if it didn't meet the dust requirements :-/

**Comment by rnicoll** [comment]
Not going to be able to find an answer on this tonight. I've tracked up and down wallet.cpp, can't see any problems, if you have any ideas let me know

**Comment by rnicoll** [comment]
Added specific unit test for the wallet fee calculation, still no sign of a problem :(

**Comment by rnicoll** [comment]
Think I may have found the problem - tests only cover calculating the fee, while below-fee transactions could still get through the wallet. Pushing this to get it off an EC2 node, but haven't finished testing, will give a thumbsup when done.

**Comment by rnicoll** [comment]
Looks good to me, now: https://chain.so/tx/DOGETEST/d9195e65bbe92a4bb63eaf267ce8d378a7d6c626a340fab24d27d24425bfb4f6

**Comment by langerhans** [comment]
Odd, this is what I see. Clean build with this PR merged in.

image

Sample transaction: https://chain.so/tx/DOGE/cb505cdf8f20041f86338041d8ef36068d5c5abe82ceab9795a6f026352aa541  

I'll have to sync in Linux to actually try and debug it.

**Comment by rnicoll** [comment]
You've got the commits from this morning included, right?

I'll try re-testing in the morning, always possible I've got versions muddled...

**Comment by langerhans** [comment]
I did, I re-checkoud out your whole PR. But I'll check that too :)

**Comment by rnicoll** [comment]
Found a few more places that references free transactions, ripped them out to be sure, but seems okay here *confused*

untitled

**Comment by langerhans** [comment]
@rnicoll It seems to behave differently on Testnet and Mainnet. On Testnet I can reproduce your results, on Mainnet I get 1 DOGE for a 0.5 DOGE send.

**Comment by rnicoll** [comment]
Thought I'd disabled the fee estimation from existing transactions, but might be it's seeing most transactions of the same size pay 1 DOGE, and using that...

**Comment by rnicoll** [comment]
Nope, txconfirmtarget is definitely disabled. I'm open to suggestions...

**Comment by langerhans** [comment]
Found the issue: I had a 1 DOGE fee set in the settings. That means this settings fully overrides any fee estimation. I'm not so sure if that is intended to be the way, but we can check on that later. I'll merge this now, so that holdup is finally out of the way.

**Pull request merged**